### PR TITLE
chore: fix console warning

### PR DESCRIPTION
**Pull request by thiagodallacqua-hpe**
## Description

Ticket: ET-6

This PR will remain as a draft until the PR at the [hew repo](https://github.com/determined-ai/hew/pull/115) gets merged.
 

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by keita-determined** [comment]
which code?

File: webui/react/src/components/ExperimentCreateModal.test.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { render, screen } from '@testing-library/react';
+import { render, screen, waitFor } from '@testing-library/react';
```
Comment (keita-determined) [inline webui/react/src/components/ExperimentCreateModal.test.tsx:1]: why do we need `waitFor`?

File: webui/react/src/components/JupyterLabModal.tsx
Comment (keita-determined) [inline webui/react/src/components/JupyterLabModal.tsx:1]: whats this change for?

**Review by keita-determined — COMMENTED** [review]
i still see warnings i listed in the ticket. not sure if i follow the changes. what are these changes for? and could you add test plans

File: webui/react/src/components/ExperimentCreateModal.tsx [hunk 2]
```diff
@@ -314,8 +314,8 @@ const ExperimentCreateModalComponent = ({
       };
       return _.isEqual(prev, newModalState) ? prev : newModalState;
     });
-    form.validateFields(requiredFields); // initial disabled state set here, gets updated later in handleFieldsChange
-  }, [experiment, trial, type, isFork, form, requiredFields]);
+    if (modalState.isAdvancedMode) form.validateFields(requiredFields); // initial disabled state set here, gets updated later in handleFieldsChange
```
Comment (keita-determined) [inline webui/react/src/components/ExperimentCreateModal.tsx:317]: qq: which console error does this code fix?

**Review by keita-determined — CHANGES_REQUESTED** [review]
still see `Instance created by `useForm` is not connected to any Form element. Forget to pass `form` prop?` in admin page. could you check all modals.

---
`Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?` too in jupyter modal.

---

For the `validateDOMNesting` issue, reaching out to the code author might be a good idea

File: webui/react/src/components/ExperimentCreateModal.test.tsx (hunk 1 above)
Comment (keita-determined) [reply webui/react/src/components/ExperimentCreateModal.test.tsx:1]: i tested without it and it worked

**Comment by keita-determined** [comment]
@thiagodallacqua-hpe i tested in private mode, so i dont think its cache issue. Could you double check

image

image

File: webui/react/src/components/ExperimentCreateModal.test.tsx [hunk 3]
```diff
@@ -65,8 +65,16 @@ describe('Create Experiment Modal', () => {
 
   it('submits a valid create experiment request', async () => {
     await setup();
+    const nameInput = screen.getByRole('textbox');
+
```
Comment (keita-determined) [inline webui/react/src/components/ExperimentCreateModal.test.tsx:69]: nit: remove newlines

**Review by keita-determined — CHANGES_REQUESTED** [review]
still see the warnings. could you ask someone to reproduce these before re-requesting the review?

https://github.com/determined-ai/determined/assets/109113616/1fee4780-cc61-4fc6-a293-a2d35b6931c8

**Comment by johnkim-det** [comment]
I am reproducing testing with local branch against latest-main API. FWIW I think I only see them once, and if I navigate away and back it doesn't show up. And then I see them again when I refresh the app.

**Comment by keita-determined** [comment]
thanks for the fixes, but unfortunately, these warnings are still reproducible, and i dont thnk the jupyterlab modal warning isn't from antd based on the warning.
I'm open to pair to reproduce it, feel free to ping me anytime.

**Pull request closed without merging**